### ISSUE TITLE
Components spring cleaning

### ISSUE THREAD

**Issue by tarelli** [openworm/geppetto-client#113]
- [ ] Do an inventory of all components inside `controls`, `widgets`, `interface`. The goal is we need to qualify whether to remove or not each of the components which depends on code quality, usage and usefulness. The question we want to answer is: if we were to build a new geppetto application today is this a component that we would use or not? Track their usage (check geppetto-vfb, geppetto-osb, geppetto-application, geppetto-hm, geppetto-netpyne, geppetto-nwbexplorer), check last commit, check code quality (ES6? Full react? is jQuery used? is it used by the [ComponentFactory](https://github.com/openworm/geppetto-client/blob/master/js/components/ComponentFactory.js)?). For instance, we probably want to remove all controls since nowadays we can just rely on an external library to do the job, e.g. Material UI
- [ ] Review the inventory, agree what we are going to remove.
- [ ] Remove the components
- [ ] Create a card to document all components that we have not removed on org.geppetto.docs
- [ ] Create a card to create a showcase for the components that we have not removed (e.g. https://github.com/openworm/geppetto-client/blob/development/js/components/interface/listViewer/showcase/ListViewerShowcase.js)

**Comment by afonsobspinto** [comment]
An initial spreadsheet can be found [here](https://docs.google.com/spreadsheets/d/1wXXNUFxkSLFflRk9f0foPRVl-9zePQw3_Vhk3aO9Iz4/edit?usp=sharing) (generated with https://github.com/afonsobspinto/GeppettoClientStats)

**Next steps:**
- [ ] Evaluate effectiveness of the script (by manually go through (some of) the values generated and evaluate them -> new column **'_reviewed_'** will be added)
- [ ] Signal relevant components

**Questions:**
1. Definition of full-react
2. Definition of relevant components  (which stats would made a component be signalled as **_to be removed_**?)

**Comment by afonsobspinto** [comment]
The [spreadsheet](https://docs.google.com/spreadsheets/d/1wXXNUFxkSLFflRk9f0foPRVl-9zePQw3_Vhk3aO9Iz4/edit#gid=960549065) was updated with:

- New columns: _Type_, _Usage_, _G._, _Geppetto._
- Fix usage: Now it as in consideration _G.addWidget_ and _GEPPETTO.ComponentFactory.addComponent_ 

In parallel a sonarcube report was generated. You can find it [here](https://drive.google.com/file/d/1zsQe_knHz0U_ZzNln39yUBbsDtMxBbmR/view?usp=sharing)

Proposal next steps:

- [ ] Join information of sonarcube to spreedsheet
- [ ] Signal relevant components

Questions:

1. Presence of false-positives like: https://github.com/openworm/geppetto-client/blob/master/js/components/interface/3dCanvas/SceneController.js#L100

**Comment by afonsobspinto** [comment]
A new [spreadsheet](https://docs.google.com/spreadsheets/d/1dJGsxASx4U2ccD7AOCAqLMPQ6xrzk2kqxX2zz4orRNo/edit?usp=sharing) was generated with:

- New columns: Applications Usage

- Fix usage: Now it as in consideration relative path imports/requires (needs to contain component name and last folder from filepath - necessary to distinguish between https://github.com/openworm/geppetto-client/blob/master/js/components/controls/mixins/Button.js and https://github.com/openworm/geppetto-client/blob/master/js/components/controls/button/Button.js for example)

**Comment by filippomc** [comment]
Closing as merged into #174
